icd/api/include/vk_utils.h

```diff
@@ -135,8 +135,8 @@ namespace utils
 // Get time in nano seconds
 VK_INLINE uint64_t GetTimeNano()
 {
-    return (static_cast<Pal::uint64>(Util::GetPerfCpuTime()) * NANOSECONDS_IN_A_SECOND) /
-        static_cast<Pal::uint64>(Util::GetPerfFrequency());
+    return static_cast<Pal::uint64>(Util::GetPerfCpuTime()) *
+        (NANOSECONDS_IN_A_SECOND / static_cast<Pal::uint64>(Util::GetPerfFrequency()));
```

Comment:
Unfortunately this function is used in multiple places to compute timings, e.g.:
```
start = GetNanoTime(); 
end = GetNanoTime();
elapsed = end - start;
```
This computes the time in nanoseconds twice while a better approach would be to store the perf counter (`GetPerfCpuTime()`), perform the substraction on the perf counter, and then do the whole ` * nanoseconds / frequency`. That would avoid the overflow in all cases while maintaining accuracy. 

So yes this fix can introduce some accuracy loss but it fixes a common use case. If you want I can investigate all the uses of this pattern and fix them at the source.

Reply:
I know it is more of a hassle, but I think the best thing is to do what you suggest - track start and end in ticks  and replace GetTimeNano() with a helper function TicksToNano() or something like that.

Reply:
hi, is the implementation of Util::GetPerfCpuTime() here ?
[https://github.com/GPUOpen-Drivers/pal/blob/dev/src/util/lnx/lnxSysUtil.cpp](https://github.com/GPUOpen-Drivers/pal/blob/dev/src/util/lnx/lnxSysUtil.cpp)
if so, Util::GetPerfCpuTime() returns current time in nanoseconds, Util::GetPerfFrequency() returns NANOSECONDS_IN_A_SECOND, the result of NANOSECONDS_IN_A_SECOND / static_cast<Pal::uint64>(Util::GetPerfFrequency()) is 1.

Reply:
To avoid future problems, I've updated the two uses of the function to use ticks, take the difference, and then use the new TicksToNano (using the original formulation).